File: api/app/Models/Elevator.php
<?php
// app/Models/Elevator.php

namespace App\Models;

use App\Models\Concerns\HasTenant;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Support\Str;

class Elevator extends Model
{
    use HasTenant, SoftDeletes;

    public const STATUS_ACTIVE  = 'active';
    public const STATUS_PASSIVE = 'passive';
    public const STATUS_FAULTY  = 'faulty';

    protected $fillable = [
        'tenant_id', 'building_id', 'code', 'name', 'type', 'brand', 'model',
        'capacity_kg', 'speed', 'floor_count', 'stop_count', 'production_year',
        'serial_number', 'tse_certificate_no', 'tse_start_date', 'tse_end_date',
        'tse_document_url', 'status', 'last_maintenance_at', 'next_maintenance_at',
        'maintenance_period', 'qr_token', 'notes', 'is_sample',
    ];

    protected function casts(): array
    {
        return [
            'speed'               => 'decimal:2',
            'tse_start_date'      => 'date',
            'tse_end_date'        => 'date',
            'last_maintenance_at' => 'datetime',
            'next_maintenance_at' => 'datetime',
            'is_sample'           => 'boolean',
        ];
    }

    protected static function booted(): void
    {
        static::creating(function (Elevator $elevator) {
            if (empty($elevator->qr_token)) {
                $elevator->qr_token = Str::lower(Str::random(32));
            }
        });
    }

    public function building(): BelongsTo
    {
        return $this->belongsTo(Building::class);
    }

    /**
     * TSE etiket rengi: green (>30g), yellow (≤30g), red (geçmiş), gray (belge yok).
     */
    public function tseLabel(): string
    {
        if (! $this->tse_end_date) {
            return 'gray';
        }
        $days = now()->startOfDay()->diffInDays($this->tse_end_date, false);
        if ($days < 0) {
            return 'red';
        }
        return $days <= 30 ? 'yellow' : 'green';
    }
}

File: api-net/Models/Elevator.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace Liftonom.Api.Models;

public class Elevator
{
    public long Id { get; set; }
    public long TenantId { get; set; }
    public long? BuildingId { get; set; }
    public string? Code { get; set; }
    public string? Name { get; set; }
    public string? Type { get; set; }
    public string? Brand { get; set; }
    public string? Model { get; set; }
    public int? CapacityKg { get; set; }
    public int? CapacityPersons { get; set; }
    public int? StopCount { get; set; }
    public string? ServedFloors { get; set; }
    public decimal? SpeedMs { get; set; }
    public string? DoorType { get; set; }
    public string? SerialNumber { get; set; }
    public string? RegistrationNo { get; set; }
    public int? ManufactureYear { get; set; }
    public DateOnly? InstallationDate { get; set; }
    public string? TseCertificateNo { get; set; }
    public DateOnly? TseStartDate { get; set; }
    public DateOnly? TseEndDate { get; set; }
    public string? TseLabelColor { get; set; }
    public string? TseLabelNote { get; set; }
    public bool HasEmergencyPhone { get; set; }
    public bool HasUps { get; set; }
    public bool HasFireSystem { get; set; }
    public bool HasEarthquakeSensor { get; set; }
    public string Status { get; set; } = "active";
    public DateTime? LastMaintenanceAt { get; set; }
    public DateTime? NextMaintenanceAt { get; set; }
    public int MaintenancePeriod { get; set; } = 30;
    public string? QrToken { get; set; }
    public string? Notes { get; set; }
    public bool IsSample { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public DateTime? DeletedAt { get; set; }

    public Building? Building { get; set; }

    [NotMapped]
    public string TseLabel
    {
        get
        {
            if (TseEndDate is null) return "gray";
            var days = TseEndDate.Value.DayNumber - DateOnly.FromDateTime(DateTime.UtcNow).DayNumber;
            if (days < 0) return "red";
            return days <= 30 ? "yellow" : "green";
        }
    }
}
